packages/gatsby-source-filesystem/src/create-remote-file-node.js

```diff
@@ -3,112 +3,283 @@ const got = require(`got`)
+  try {
+    const response = await requestRemoteNode(url, headers, tmpFilename, filename)
+    // Save the response headers for future requests.
+    cache.set(cacheId(url), response.headers)
 
     // If the status code is 200, move the piped temp file to the real name.
+    if (response.statusCode === 200) {
+      await fs.move(tmpFilename, filename, { overwrite: true })
     // Else if 304, remove the empty response.
-    responseStream.on(`response`, response => {
-      statusCode = response.statusCode
-      responseHeaders = response.headers
-    })
+    } else {
+      await fs.remove(tmpFilename)
+    }
 
-    responseStream.on(`end`, response => {
-      if (responseError) return
-
-      // Save the response headers for future requests.
-      cache.set(cacheId(url), responseHeaders)
-      if (statusCode === 200) {
-        fs.moveSync(tmpFilename, filename, { overwrite: true })
-      } else {
-        fs.removeSync(tmpFilename)
-      }
-
-      // Create the file node and return.
-      createFileNode(filename, {}).then(fileNode => {
-        // Override the default plugin as gatsby-source-filesystem needs to
-        // be the owner of File nodes or there'll be conflicts if any other
-        // File nodes are created through normal usages of
-        // gatsby-source-filesystem.
-        createNode(fileNode, { name: `gatsby-source-filesystem` })
-        resolve(fileNode)
-      })
+    // Create the file node.
+    const fileNode = await createFileNode(filename, {})
+
+    // Override the default plugin as gatsby-source-filesystem needs to
+    // be the owner of File nodes or there'll be conflicts if any other
+    // File nodes are created through normal usages of
+    // gatsby-source-filesystem.
+    createNode(fileNode, { name: `gatsby-source-filesystem` })
+
+    return fileNode
+  } catch (err) {
+    // ignore
+  }
+  return null
+}
+
+
+/**
+ * Index of promises resolving to File node from remote url
+ */
+const processingCache = {}
+/**
+ * pushTask
+ * --
+ * pushes a task in to the Queue and the processing cache
+ *
+ * Promisfy a task in queue
+ * @param {CRFNPayload} task
+ * @return {Promise<Object>}
+ */
+const pushTask = (task) => new Promise((resolve, reject) => {
+  queue
+    .push(task)
+    .on(`finish`, (task) => {
+      resolve(task)
+    })
+    .on(`failed`, () => {
+      resolve()
     })
-  }))
+})
+
+
+/***************
+ * Entry Point *
+ ***************/
+
+/**
+ * createRemoteFileNode
+ * --
+ *
+ * Download a remote file
+ * First checks cache to ensure duplicate requests aren't processed
+ * Then pushes to a queue
+ *
+ * @param {CRFNPayload} options
+ * @return {Promise<Object>}                  Returns the created node
+ */
+module.exports = ({ url, store, cache, createNode, auth = {} }) => {
+  // Check if we already requested node for this remote file
+  // and return stored promise if we did.
+  if (processingCache[url]) {
+    return processingCache[url]
+  }
+
+
+
```

Comment:
just cosmetic - can you remove multi empty lines?

Reply:
Sure thing